src/main/java/chalmers/pimp/model/ModelImpl.java

```diff
@@ -218,6 +221,15 @@ public void selectedToolReleased(MouseStatus mouseStatus) {
   }
 
   @Override
+  public void replaceLayer(int index, ILayer layer) {
+    if (canvas.getAmountOfLayers() <= index) {
+      return;
+    }
```

Comment:
Use the method Canvas.verifyIndexedLayerExistence(int). Should we make another method with params layer, layer (to replace first layer with second supplied layer?). We have both kinds for alot of methods in Canvas